rebasebot/bot.py

```diff
@@ -149,11 +149,19 @@ def _do_rebase(gitwd, source, dest, source_repo, tag_policy):
                             "--ancestry-path", f"{merge_base}..dest/{dest.branch}")
     logging.info("Picking commits: \n%s", commits)
 
+    go_mod_commits = []
+
     for commit in commits.splitlines():
         # Commit contains the message for logging purposes,
         # trim on the first space to get just the commit sha
         sha, commit_message = commit.split(" - ")
 
+        if update_go_modules:
+            if commit_message.find("UPSTREAM: <carry>: Updating and vendoring go modules "
+                "after an upstream rebase"):
+                commit_message.replace("<carry>", "<drop>")
+                go_mod_commits.append(sha)
+
```

Comment:
So this will prune old go mod updates? Nice! Can you add a comment explaining that this is what this is doing

Reply:
> Nice, how have you tested this?
TBH, I have not yet. I haven't figured out a way to test this, since the rebasebot always rewrites the `.rebase` folder and I can't find a repo with more than 2 of these commits. Do you have any suggestions?

This piece of code will append commit sha's into a list IFF the commit message corresponds to the message of go_mod_update function and then replaces the `<carry>` flag with `<drop>` flag to indicate that it is something that we don't want to rebase. After all that, we just cherry pick these commits and delete them. Will add description to the code as well.

Reply:
Great, thanks :)